### PR TITLE
DOC-2524: New `tinycomments` option: Comments can be made and read, but the editor is in a readonly state.

### PR DESCRIPTION
**Pull request by kemister85**
Ticket: DOC-2524
Supports: TINY-11113

Site: [Staging branch](http://docs-feature-74-doc-2524doc-2539.staging.tiny.cloud/docs/tinymce/latest/comments-access-options/)

Changes:

Created new `comments-readonly-mode` demo.
* `live-demos/comments-readonly-mode/index.html`
* `live-demos/comments-readonly-mode/index.js`

Created new `comments-ui-demo`
* `live-demos/comments-ui-mode/index.html`
* `live-demos/comments-ui-mode/index.js`

Added new `comments-options.adoc to:
* `nav.adoc`
* Included new `partials/configuration/comments-tinycomments_access.adoc`

Pre-checks:
- [x] Branch prefixed with `feature/<version>/`, `hotfix/<version>/`, `staging/<version>/`, or `release/<version>/`.
- [x] `modules/ROOT/nav.adoc` has been updated `(if applicable)`.

Review:
- [x] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -303,6 +303,7 @@
 **** xref:comments-callback-mode.adoc[Callback mode]
 **** xref:comments-embedded-mode.adoc[Embedded mode]
 **** xref:comments-toolbars-menus.adoc[Toolbar buttons and menu items]
+**** xref:comments-options.adoc[Ui Options]
```
Comment (ShiridiGandham) [inline modules/ROOT/nav.adoc:306]:
```suggestion
**** xref:comments-options.adoc[Access Options]
```

File: modules/ROOT/pages/comments-options.adoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
+:navtitle: Comments Options
+:description: TinyMCE Comments plugin options.
+:keywords: comments, commenting, tinycomments
+:pluginname: Comments
+:plugincode: comments
+:pluginminimumplan: enterprise
+
+== Options
```
Comment (ShiridiGandham) [inline modules/ROOT/pages/comments-options.adoc:8]:
```suggestion
== Access Options
```

File: modules/ROOT/pages/comments-options.adoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+The following configuration options affect the behavior of the {pluginname} plugin.
```
Comment (ShiridiGandham) [inline modules/ROOT/pages/comments-options.adoc:10]:
```suggestion
The following configuration option affects the behavior of the {pluginname} plugin.
```

File: modules/ROOT/pages/comments-options.adoc [hunk 4]
```diff
@@ -0,0 +1,12 @@
+:navtitle: Comments Options
+:description: TinyMCE Comments plugin options.
+:keywords: comments, commenting, tinycomments
+:pluginname: Comments
+:plugincode: comments
+:pluginminimumplan: enterprise
```
Comment (ArvinJ-H) [inline modules/ROOT/pages/comments-options.adoc:6]: should this be `essential`?
https://www.tiny.cloud/pricing/

Comment (MitchC1999) [reply modules/ROOT/pages/comments-options.adoc:6]: I think we should make a ticket to remove all references to this variable. We no longer include any plan details in the documentation so this variable is no longer used.

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+When `tinycomments_access` is set to **undefined** or an **empty string**, the editor will remain in read-only mode, and the comments will be in read-only mode. This means that users cannot edit the content or add comments.
+
+[cols="1,1,2", options="header"]
+|===
+|Editor Readonly |Tinycomments_access |Result
+|true |not specified |readonly editor, readonly comments
+|false |not specified |Default behaviour, full editing rights to the comments and editor
+|===
+
```
Comment (ShiridiGandham) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:39]:
```suggestion
=== `tinycomments_access` option is either configured with an empty string or not configured at all,and `readonly` set to `true`

When `tinycomments_access` is not configured or set to an empty string and `read-only` is set to true, the editor and comments will be in read-only mode. In this state, users cannot edit content or add comments.

[cols="1,1,2", options="header"]
|===
|Editor Readonly |Tinycomments_access |Result
|true |not specified |readonly editor, readonly comments
|false |not specified |Default behaviour, full editing rights to the comments and editor
|===
=== Example:

[source,javascript]
----
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'tinycomments',
  toolbar: 'addcomment showcomments',
  readony: true, // set the editor to read-only mode
  tinycomments_access: '' 
});
----



```

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+*Type:* `+String+`
+
+*Possible values:* `'comment'`
+
+This option controls the editor's read-only state and comment functionality.
+
+[IMPORTANT]
+====
+* When `tinycomments_access` is set to `comment`, the editor will automatically switch to xref:editor-important-options.adoc#readonly[Readonly Mode].
+* Default behavior allows full editing access to both the editor and comments.
```
Comment (ShiridiGandham) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:13]:
```suggestion
* Default behavior, when the `tinycomments_access` option is either not configured or set to an empty string, allows full editing access to both the editor and the comments.
```

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+When `tinycomments_access` is set to `comment` and `readonly` is set to `true`, the editor **will remain** in read-only mode, but the comments will be in **comment-only** mode. This means that users can add and delete comments, but they cannot edit the editor content.
+
+[cols="1,1,2", options="header"]
+|===
+|Editor Readonly |Tinycomments_access |Result
+|true |comment |comment only mode, readonly editor, can add/delete new comments
+|false |comment |comment only mode, readonly editor, can add/delete new comments
+|===
+
```
Comment (ShiridiGandham) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:52]:
```suggestion
== `tinycomments_access` set to `'comment'` and `readonly` set to `true`

When `tinycomments_access` is set to `comment` and `readonly` is set to `true`, the editor **will remain** in read-only mode, but the comments will be in **comment-only** mode. This means that users can add and delete comments, but they cannot edit the editor content.

[cols="1,1,2", options="header"]
|===
|Editor Readonly |Tinycomments_access |Result
|true |comment |The editor is read-only, but users can add or delete new comments.
|false |comment |The editor is read-only, but users can add or delete new comments.
|===
=== Example:
[source,javascript]
----
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'tinycomments',
  toolbar: 'addcomment showcomments',
  readony: true, // set the editor to read-only mode
  tinycomments_access: 'comment' 
});
----
```

File: modules/ROOT/examples/live-demos/comments-readonly-mode/index.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+tinymce.init({
+  selector: "textarea#comment-readonly-ui-mode",
+  plugins: [
+    "tinycomments", "advlist", "anchor", "autolink", "charmap", "code", "fullscreen",
+    "help", "image", "insertdatetime", "link", "lists", "media",
+    "preview", "searchreplace", "table", "visualblocks",
+  ],
+  toolbar: "addcomment showcomments | undo redo | styles | bold italic underline strikethrough | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link image",
+  tinycomments_mode: 'embedded',
+  tinycomments_access: '',
```
[the remark this answers is not in the document]
Comment (kemister85) [reply modules/ROOT/examples/live-demos/comments-readonly-mode/index.js:12]: Good point, thanks @MitchC1999

File: modules/ROOT/examples/live-demos/comments-ui-mode/index.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    "tinycomments", "advlist", "anchor", "autolink", "charmap", "code", "fullscreen",
+    "help", "image", "insertdatetime", "link", "lists", "media",
+    "preview", "searchreplace", "table", "visualblocks",
+  ],
+  toolbar: "addcomment showcomments | undo redo | styles | bold italic underline strikethrough | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link image",
+  tinycomments_mode: 'embedded',
+  tinycomments_access: 'comment',
+  setup: (ed) => {
+    editor = ed;
+  }
```
Comment (shanmen-tiny) [inline modules/ROOT/examples/live-demos/comments-ui-mode/index.js:15]:
```suggestion
 ```

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'tinycomments',
+  toolbar: 'addcomment showcomments',
+  readony: true, // set the editor to read-only mode
+});
+----
+
+=== `tinycomments_access` set to undefined or an empty string and `readonly` set to `true`
+
+When `tinycomments_access` is set to **undefined** or an **empty string**, the editor will remain in read-only mode, and the comments will be in read-only mode. This means that users cannot edit the content or add comments.
```
Comment (shanmen-tiny) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:30]:
```suggestion
=== `tinycomments_access` is not configured and `readonly` set to `true`

When `tinycomments_access` is not configured, the editor will remain in read-only mode, and the comments will be in read-only mode. This means that users cannot edit the content or add comments.
```
The option could be not configured, could be any values that's invalid and the `readonly: true` will still overwrite the editor state. Maybe just not configured or invalid?

**Review by shanmen-tiny — APPROVED** [review]
🙇‍♂️

File: modules/ROOT/examples/live-demos/comments-readonly-mode/index.html [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  </textarea>
+  <!-- <button id="toggle-readonly-mode">Read-Only Mode Enabled</button> -->
+  <button id="toggle-readonly-mode" style="margin-top: 1em; padding: 0.5em 1em; background-color: #17224f; color: #fff; border: none; border-radius: .2rem .2rem 0 0;">Read-Only Mode Enabled</button>
```
Comment (shanmen-tiny) [inline modules/ROOT/examples/live-demos/comments-readonly-mode/index.html]:
```suggestion
```

**Comment by kemister85** [comment]
@MitchC1999 @shanmen-tiny  Can you review last [Commit](https://github.com/tinymce/tinymce-docs/pull/3461/commits/0fed7b28830a318ef5064970271ca29669f1955b), I think in this case we can remove basic config example as we have two examples below, and having a config with set to `tinycomments_access: ' '` as discussed is not optional.

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+*Type:* `+String+`
+
+*Possible values:* `'comment'`
+
+This option controls the editor's read-only state and comment functionality.
+
+[IMPORTANT]
+====
+* When `tinycomments_access` is set to `comment`, the editor will automatically switch to xref:editor-important-options.adoc#readonly[Readonly Mode].
+* Default behavior, when the `tinycomments_access` option is either not configured or set to an empty string, allows full editing access to both the editor and the comments.
```
Comment (MitchC1999) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:13]:
```suggestion
* When `tinycomments_access` is set to `comment`, the editor will automatically switch to xref:editor-important-options.adoc#readonly[Readonly Mode].
```

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 13]
```diff
@@ -0,0 +1,63 @@
+[[tinycomments-access]]
+== `tinycomments_access`
+
+*Type:* `+String+`
+
+*Possible values:* `'comment'`
+
+This option controls the editor's read-only state and comment functionality.
```
Comment (MitchC1999) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:8]:
```suggestion
This option sets the editor to the readonly mode, while still allowing comments to be made. 
```

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+[IMPORTANT]
+====
+* When `tinycomments_access` is set to `comment`, the editor will automatically switch to xref:editor-important-options.adoc#readonly[Readonly Mode].
+* Default behavior, when the `tinycomments_access` option is either not configured or set to an empty string, allows full editing access to both the editor and the comments.
+====
+
+=== `tinycomments_access` _is set to `'comment'` and `readonly` set to `true`_
+
+When `tinycomments_access` is set to `comment` and `readonly` is set to `true`, the editor **will remain** in read-only mode, but the comments will be in **comment-only** mode. This means that users can add and delete comments, but they cannot edit the editor content.
```
Comment (MitchC1999) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:18]:
```suggestion
When `tinycomments_access` is set to `comment`, the editor will be in xref:editor-important-options.adoc#readonly[Readonly mode], but the comments will be in a **comment-only** mode. This means that users can add and delete comments, but they cannot edit the editor content.
```

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+[cols="1,1,2", options="header"]
+|===
+|Editor Readonly |Tinycomments_access |Result
+|true |comment |The editor is read-only, but users can add or delete new comments.
+|false |comment |The editor is read-only, but users can add or delete new comments.
+|===
```
Comment (MitchC1999) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:25]:
```suggestion
```
These are the same thing, so don't include it.

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+.Example
+[source,javascript]
+----
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'tinycomments',
+  toolbar: 'addcomment showcomments',
+  readony: true, // set the editor to read-only mode
```
Comment (MitchC1999) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:34]:
```suggestion
```
Unnecessary, `tinycomments_access: "comment"` sets the editor to readonly already, so the integrator doesn't need to use the `readonly` option.

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+=== `tinycomments_access` _is not configured and `readonly` set to `true`_
+
+When `tinycomments_access` is not configured, the editor will remain in read-only mode, and the comments will be in read-only mode. This means that users cannot edit the content or add comments.
+
+[cols="1,1,2", options="header"]
+|===
+|Editor Readonly |Tinycomments_access |Result
+|true |not specified |readonly editor, readonly comments
+|false |not specified |Default behavior, full editing rights to the comments and editor
+|===
```
Comment (MitchC1999) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:50]:
```suggestion
=== Comments with `readonly` set to `true`

When not using the `tinycomments_access` option, the comments sidebar will have a different state depending on the use of the xref:editor-important-options.adoc#readonly[`readonly` option].

With neither xref:editor-important-options.adoc#readonly[Readonly mode] nor `tinycomments: "comment"` configured, both the editor and the comments sidebar will be in their default state. By default, comments can be both viewed and added, and the editor content is editable.

With xref:editor-important-options.adoc#readonly[Readonly mode] enabled, the comments sidebar will be in a **view-only** mode, meaning all comments can be viewed and navigated, but no new comments can be added. Also, in the **readonly** state, the editor will still not be editable.

```

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+
+.Example
+[source,javascript]
+----
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'tinycomments',
+  toolbar: 'addcomment showcomments',
```
Comment (ShiridiGandham) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:56]:
```suggestion
  plugins: 'tinycomments',
  tinycomments_mode: 'embedded',
  toolbar: 'addcomment showcomments',
```

File: modules/ROOT/partials/configuration/comments-tinycomments_access.adoc [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+*Possible values:* `'comment'`
+
+This option sets the editor to the readonly mode, while still allowing comments to be made. 
+
+[IMPORTANT]
+====
+* When `tinycomments_access` is set to `comment`, the editor will automatically switch to xref:editor-important-options.adoc#readonly[Readonly Mode].
+====
+
+=== `tinycomments_access` _is set to `'comment'` and `readonly` set to `true`_
```
Comment (MitchC1999) [inline modules/ROOT/partials/configuration/comments-tinycomments_access.adoc:15]:
```suggestion
=== Comments with `tinycomments_access` set to `'comment'`
```

**Review by MitchC1999 — APPROVED** [review]

File: modules/ROOT/examples/live-demos/comments-ui-mode/index.html [hunk 20]
```diff
@@ -0,0 +1,38 @@
+<div id="tiny-comments-ui-mode">
+  <textarea id="comments-ui-mode" style="width: 100%; height: 500px;">
+    <h2>Welcome to the New Tiny Comments Mode!</h2>
+    <p>This demo showcases the new Comments UI Mode. You can add or delete comments on the content, but the content itself is locked for editing.</p>
+    <ol>
+      <li>Highlight the text you want to comment on.</li>
+      <li>Click the <em>Add Comment</em> icon in the toolbar.</li>
+      <li>Type your comment into the text field in the Comment sidebar.</li>
+      <li>Click <strong>Comment</strong> to attach the comment to the text.</li>
+      <li>To delete a comment, click the comment bubble and choose <strong>Delete Comment</strong>.</li>
```
Comment (tiny-ben-tran) [inline modules/ROOT/examples/live-demos/comments-ui-mode/index.html:10]:
```suggestion
      <li>To delete a comment, click the comment options and choose <strong>Delete Comment</strong>.</li>
```

**Review by tiny-ben-tran — APPROVED** [review]

**Pull request merged**